app/modules/import-export/assets/js/admin.js

```diff
@@ -1,19 +1,214 @@
+		}
+		revertButton.addEventListener( 'click', this.revertBtnOnClick.bind( this ) );
+		this.maybeAddRevertBtnMargin( revertButton );
+	}
+
+	/**
+	 * MaybeAddRevertBtnMargin
+	 *
+	 * @param {HTMLAnchorElement} revertButton
+	 */
+	maybeAddRevertBtnMargin( revertButton ) {
+		const referrerKitId = new URLSearchParams( revertButton.href ).get( 'referrer_kit' );
+		if ( ! referrerKitId ) {
+			return;
+		}
+		revertButton.style.marginBottom = this.calculateMargin( revertButton );
+		this.scrollToBottom();
+	}
+
+	/**
+	 * CalculateMargin
+	 *
+	 * @param {HTMLAnchorElement} revertButton
+	 *
+	 * @return {string}
+	 */
+	calculateMargin( revertButton ) {
+		const adminBar = document.getElementById( 'wpadminbar' );
+		const adminBarHeight = adminBar ? adminBar.offsetHeight : 0;
+
+		const revertKitHeight = revertButton.parentElement.offsetHeight;
+
+		return (
+			document.body.clientHeight -
+			adminBarHeight -
+			revertKitHeight -
+			document.getElementById( 'wpfooter' ).offsetHeight -
+			15 // Extra margin at the top
+		) + 'px';
+	}
+
+	/**
+	 * Scroll to the bottom of the page
+	 */
+	scrollToBottom() {
+		setTimeout( () => {
+			window.scrollTo( 0, document.body.scrollHeight );
+		} );
+	}
+
+	/**
+	 * RevertBtnOnClick
+	 *
+	 * @param {Event} event
+	 */
+	revertBtnOnClick( event ) {
 		event.preventDefault();
 
+		const kitNameToRemove = this.getKitNameToRemove();
+
 		elementorCommon.dialogsManager.createWidget( 'confirm', {
-			headerMessage: __( 'Sure you want to make these changes?', 'elementor' ),
-			message: __( 'Removing assets or changing your site settings can drastically change the look of your website.', 'elementor' ),
+			headerMessage: __( 'Are you sure?', 'elementor' ),
+			message: __( 'Removing ', 'elementor' ) + kitNameToRemove + __( ' will permanently delete changes made to the Kit\'s content and site settings', 'elementor' ),
 			strings: {
-				confirm: __( 'Yes', 'elementor' ),
-				cancel: __( 'No, go back', 'elementor' ),
+				confirm: __( 'Delete', 'elementor' ),
+				cancel: __( 'Cancel', 'elementor' ),
 			},
-			onConfirm() {
-				location.href = revertButton.href;
+			onConfirm: () => this.onRevertConfirm( event.target, kitNameToRemove ),
+		} ).show();
+	}
+
+	onRevertConfirm( revertBtn, kitNameToRemove ) {
+		const referrerKitId = new URLSearchParams( revertBtn.href ).get( 'referrer_kit' );
+
+		this.saveToCache( referrerKitId ?? '', kitNameToRemove );
+
+		location.href = revertBtn.href;
+	}
+
+	maybeShowReferrerKitDialog() {
+		const { referrerKitId } = this.getDataFromCache();
+		if ( undefined === referrerKitId ) {
+			return;
+		}
+
+		if ( 0 === referrerKitId.length ) {
+			this.createKitDeletedWidget( {
+				message: __( 'Try a different Kit or build your site from scratch.', 'elementor' ),
+				strings: {
+					confirm: __( 'OK', 'elementor' ),
+					cancel: __( 'Kit Library', 'elementor' ),
+				},
+				onCancel: () => {
+					location.href = elementorAppConfig.base_url + '#/kit-library';
```

Comment:
shouldn't the URLs come from the backend?

Reply:
the base is coming from the backend, only `kit-library` no sense sending this from the backend

Reply:
IDK, seems like there are many URLs here:

https://github.com/elementor/elementor/blob/9fcaebbf68f755acf89d743e0b2234aac8cc4d5f/app/app.php#L87-L99